config/src/config.rs

```diff
@@ -147,8 +147,40 @@ pub struct IndexerConfig {
 
 #[derive(Debug, Deserialize)]
 #[cfg_attr(test, derive(PartialEq))]
-pub struct DatabaseConfig {
-    pub postgres_url: Url,
+#[serde(untagged)]
+pub enum DatabaseConfig {
+    PostgresUrl {
+        postgres_url: Url,
+    },
+    PostgresVars {
+        host: String,
+        port: Option<u16>,
+        user: String,
+        password: Option<String>,
+        database: String,
+    },
+}
+impl DatabaseConfig {
+    pub fn get_formated_postgres_url(&self) -> Url {
+        match self {
+            DatabaseConfig::PostgresUrl { postgres_url } => postgres_url.clone(),
+            DatabaseConfig::PostgresVars {
+                host,
+                port,
+                user,
+                password,
+                database,
+            } => {
+                let postgres_url_str = format!("postgres://{}@{}/{}", user, host, database);
+                let mut postgres_url = Url::parse(&postgres_url_str).unwrap();
```

Comment:
Could you use expect and add a nice message here like "Failed to parse database_url".